tests/main/test_openai_wrapper.py

```diff
@@ -121,9 +359,744 @@ def test_predict_call(self):
+                    top_p=openai_model.top_p,
+                    frequency_penalty=openai_model.frequency_penalty,
+                    presence_penalty=openai_model.presence_penalty,
+                    stop=openai_model.stop
+                )
+
+    @pytest.mark.skipif(sys.version_info.minor <= 6,
+                        reason='Openai not supported for older versions')
+    def test_predict_call_str(self):
+        # test creating the OpenaiWrapperModel and
+        # calling the predict function
+        api_type = 'azure'
+        api_base = 'https://mock.openai.azure.com/'
+        api_version = '2023-03-15-preview'
+        api_key = 'mock'
+        context = ''
+        questions = 'How to convert 10^9/l to liter?'
+        is_openai_1_0 = False
+        if not hasattr(openai, 'OpenAI'):
+            mock_function = 'openai.ChatCompletion.create'
+        else:
+            mock_function = 'openai.resources.chat.completions.Completions.create'
+            is_openai_1_0 = True
+
+        expected_model = 'gpt-4-32k'
+        mock_result = self.create_mock_result('mock', is_openai_1_0)
+
+        openai_model = OpenaiWrapperModel(
+            api_type, api_base, api_version, api_key)
+
+        # mock the openai create function
+        with patch(mock_function) as mock_create:
+            # wrap return value in mock class with read method
+            mock_create.return_value = mock_result
+            context = {}
+
+            # Only prompt
+            test_data = questions
+            openai_model.predict(context, test_data)
+            if is_openai_1_0:
+                mock_create.assert_called_with(
+                    model=expected_model,
+                    messages=[
+                        {'role': 'user', 'content': questions}
+                    ],
+                    temperature=openai_model.temperature,
+                    max_tokens=openai_model.max_tokens,
+                    top_p=openai_model.top_p,
+                    frequency_penalty=openai_model.frequency_penalty,
+                    presence_penalty=openai_model.presence_penalty,
+                    stop=openai_model.stop
+                )
+            else:
+                mock_create.assert_called_with(
+                    engine=expected_model,
+                    messages=[
+                        {'role': 'user', 'content': questions}
+                    ],
+                    temperature=openai_model.temperature,
+                    max_tokens=openai_model.max_tokens,
+                    top_p=openai_model.top_p,
+                    frequency_penalty=openai_model.frequency_penalty,
+                    presence_penalty=openai_model.presence_penalty,
+                    stop=openai_model.stop
+                )
+
+    @pytest.mark.skipif(sys.version_info.minor <= 6,
+                        reason='Openai not supported for older versions')
+    def test_predict_call_list_of_str(self):
```

Comment:
it looks like there is a lot of repeated code in the test cases - for example, this test and the one below can just re-use a helper function or be parameterized tests

Reply:
I wrote it like this so that if some tests fail we can easily see what went wrong

Reply:
I did write a helper function to provide the expected response instead of writing it in every function

Reply:
take a look at some of these parameterized tests for an example:

https://github.com/microsoft/ml-wrappers/blob/main/tests/main/test_image_model_wrapper.py#L255

https://github.com/microsoft/ml-wrappers/blob/main/tests/main/test_predictions_wrapper.py#L42

that way we could re-use code - or you can just call helper functions to refactor out common code